Better error message for missing let

jaydorsey opened the pull request:
I've run into this a number of times, and helped developers triage this
as well. This is meant to improve the developer experience by providing
an improved error message when we forget to configure a let

The original error message is actually a `NoMethodError`/`undefined method`
and it's not always immediately clear what the solution is, particularly if 
you're new to rspec/rswag

Here's an example error (the real one that prompted me to create this PR)

![image](https://user-images.githubusercontent.com/191564/128745129-4fe14ffb-d7e6-44dd-9435-1a07dc435ca8.png)

And the new error message:

![image](https://user-images.githubusercontent.com/191564/128745596-3f827f3b-6a36-4b5d-990f-b5ba9ff6f6d1.png)

I have a number of other small changes like this that I can contribute as well
if these types of pull requests are welcome.

jaydorsey committed:
Better error message for missing let

I've run into this problem a number of times. The original error message
is actually a `NoMethodError` and it's not always immediately clear what
the solution is, particuarly if you're new to rspec or rswag

Wanted to see if this kind of behavior is something that the rswag team
would be interested in adopting. I have a few other of these in mind
(will do as small PRs, with tests)

BookOfGreg commented on rswag-specs/lib/rswag/specs/request_factory.rb:
```diff
@@ -194,12 +194,36 @@ def build_form_payload(parameters, example)
+
+      def message
+        <<~MSG
+          Missing parameter #{body_param}
```
```suggestion
          Missing parameter '#{body_param}'
```

jaydorsey replied:
I updated with this suggestion & rebased off of a more recent master branch to fix the failing specs

jaydorsey committed:
Add single quotes around parameter name

jaydorsey commented:
@BookOfGreg I updated with both fo your suggestions, and fixed the failing specs by doing a rebase/update. Let me know if you want any additional changes
